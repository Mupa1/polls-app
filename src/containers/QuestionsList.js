import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import getQuestions from '../api';

import QuestionsCard from '../components/QuestionsCard';

const QuestionsList = () => {
  const dispatch = useDispatch();
  const questions = useSelector((state) => state.questions.questions);

  useEffect(() => {
    dispatch(getQuestions());
  }, []);

  return (
    <section>
      {questions.length ? (
        <>
          <h1>Questions</h1>
          <div>
            {questions.map((question, index) => (
              <div key={question.url}>
                <QuestionsCard
                  title={question.question}
                  published_at={question.published_at}
                  choices={questions[index].choices.length}
                  url={question.url}
                />
              </div>
            ))}
          </div>
        </>
      ) : (
        <div>Loading questions ...</div>
      )}
    </section>
  );
};

export default QuestionsList;
