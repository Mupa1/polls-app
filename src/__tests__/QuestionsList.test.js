import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import store from '../redux/store';
import QuestionsList from '../containers/QuestionsList';

describe('QuestionsList component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <Provider store={store}>
        <QuestionsList />
      </Provider>, div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <QuestionsList />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
