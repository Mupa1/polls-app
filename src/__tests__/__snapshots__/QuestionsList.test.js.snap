// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionsList component matches snapshot 1`] = `
<section
  className="questions-list-view"
>
  <div>
    Loading questions ...
  </div>
</section>
`;
